login/src/main/java/com/fightpandemics/login/ui/LoginViewModel.kt

```diff
@@ -136,12 +178,117 @@ class LoginViewModel @Inject constructor(
+                .collect {
+                    when (it) {
+                        is Result.Loading -> currentLocation(true, null, null)
+                        is Result.Success -> currentLocation(
+                            false,
+                            null,
+                            it.data as com.fightpandemics.core.data.model.userlocation.Location
+                        )
+                        is Result.Error -> currentLocation(true, it, null)
+                    }
+                }
+        }
+    }
+
+//     Search for location from API using user input
+    @FlowPreview
+    fun searchLocation() {
+        searchlocationJob?.cancel()
+        searchlocationJob = viewModelScope.launch {
+            searchQuery
+                .debounce(300)
+                .filter { return@filter it.isNotEmpty() && it.length >= LEN_FOR_SUGGESTIONS }
+                .distinctUntilChanged()
+                .map { it.trim() }
+                .flatMapLatest { locationPredictionsNameAndIdUseCase(it) }
+                .conflate()
+                .collect {
+                    when (it) {
+                        is Result.Success -> {
+                            _searchLocationState.value =
+                                it.data as MutableList<Prediction>
+                        }
+                        is Result.Error -> Timber.e(it.toString())
+                        is Result.Loading -> Timber.e("LOADING...")
+                    }
+                }
+        }
+    }
+
+    private fun currentLocation(
+        loading: Boolean,
+        error: Result.Error?,
+        currentLocation: com.fightpandemics.core.data.model.userlocation.Location?
+    ) {
+
+        if (currentLocation != null) {
+            completeProfileLocation = com.fightpandemics.core.data.model.login.Location(
+                currentLocation.address ?: "",
+                currentLocation.city ?: "",
+                listOf(
+                    currentLocation.coordinates?.get(0) ?: 0.0,
+                    currentLocation.coordinates?.get(1) ?: 0.0
+                ),
+                currentLocation.country ?: "",
+                currentLocation.state ?: ""
+            )
+        }
+        _currentLocationState.value =
+            UserLocationViewState(loading, error, "${currentLocation?.address}")
+    }
+
+    fun getDetails(placeId: String?) {
+        Timber.i("Debug: my placeId is $placeId")
+        viewModelScope.launch {
+            locationDetailsUseCase.invoke(placeId).collect {
+                when (it) {
+                    is Result.Success -> {
+                        val locationDetails = it.data as com.fightpandemics.core.data.model.userlocationdetails.Location
+                        Timber.i("Debug: my location prediction details $locationDetails")
+                        completeProfileLocation =
+                            com.fightpandemics.core.data.model.login.Location(
```

Comment:
See comment above. If the fields need to be nullable, maybe there could be a constructor of `Location` that takes another location and does these checks. This would avoid these checks in multiple places